pybutton/response.py

```diff
@@ -3,39 +3,64 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+from .request import urlparse
+from .request import parse_qs
+
 
 class Response(object):
-    '''The Response class wraps a return value (dict) from an API call.
+    '''The Response class wraps the returned values from an API call.
 
-    It exposes all keys in the underlying response as attributes on the
-    instance.
+    It exposes the response data via the `data` method and cursors for
+    pagination via the `next`/`prev` methods.
 
     Args:
-        attrs (dict): The underlying response value from an API call
+        meta (dict): The metadata from an API call
+        data (dict or array<dict>): The response elements from an API call
 
     Attributes:
         * (*): All keys in `attrs` will be exposed as attributes of an instance
 
     '''
 
-    def __init__(self, attrs):
-        self.attrs = attrs
-
-    def to_dict(self):
-        '''Return the raw response received from the server'''
+    def __init__(self, meta, response_data):
+        self.meta = meta
+        self.response_data = response_data
 
-        return self.attrs
+    def data(self):
+        '''Return the raw response element(s) received from the server.
+           May be a single dict or an array of dicts.
+        '''
+        return self.response_data
 
-    def __getattr__(self, attr):
-        '''Proxy attribute lookups on an instance down to the response'''
+    def next(self):
+        '''For paginated responses, returns the url used to fetch
+            the next elements.
+        '''
+        return self._format_cursor(self.meta.get('next'))
 
-        return self.attrs.get(attr)
+    def prev(self):
+        '''For paginated responses, returns the url used to fetch
+            the previous elements.
+        '''
+        return self._format_cursor(self.meta.get('prev'))
 
     def __repr__(self):
         values = []
 
-        if self.attrs:
-            for k, v in self.attrs.items():
+        if isinstance(self.response_data, dict):
+            for k, v in self.response_data.items():
                 values = values + ['{0}: {1}'.format(k, v)]
+            return '<class pybutton.Response {0}>'.format(', '.join(values))
+        elif isinstance(self.response_data, list):
+            return '<class pybutton.Response [{0} elements]>'.format(
+                len(self.response_data)
+            )
+        else:
+            return '<class pybutton.Response>'
+
+    def _format_cursor(self, cursor_url):
+        if cursor_url:
+            query_string = urlparse(cursor_url)[4]
```

Comment:
Can we add more sanity checks around these list look-ups?  I'd like to make sure the API can start returning botched URLs and this client quietly falls back to retuning `None`